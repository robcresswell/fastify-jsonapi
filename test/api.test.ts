import { it, describe, beforeEach, afterEach, expect } from 'vitest';
import { createTestServer, TestServer } from './helpers.js';

describe('api', () => {
  let server: TestServer;

  beforeEach(async () => {
    server = await createTestServer();
  });

  afterEach(async () => {
    await server.close();
  });

  it('returns a well formed api', async () => {
    await server.ready();

    expect(server.swagger()).toMatchSnapshot();
  });
});
