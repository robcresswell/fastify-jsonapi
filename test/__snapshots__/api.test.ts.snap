// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`api > returns a well formed api 1`] = `
{
  "components": {
    "schemas": {},
  },
  "info": {
    "title": "@fastify/swagger",
    "version": "9.4.0",
  },
  "openapi": "3.1.1",
  "paths": {
    "/empty": {
      "get": {
        "responses": {
          "200": {
            "description": "Default Response",
          },
        },
      },
    },
    "/items": {
      "get": {
        "parameters": [
          {
            "in": "query",
            "name": "sort",
            "required": false,
            "schema": {
              "default": "name",
              "enum": [
                "name",
                "-name",
                "createdAt",
                "-createdAt",
              ],
              "type": "string",
            },
          },
          {
            "in": "query",
            "name": "page[size]",
            "required": false,
            "schema": {
              "maximum": 100,
              "minimum": 1,
              "type": "integer",
            },
          },
          {
            "in": "query",
            "name": "page[after]",
            "required": false,
            "schema": {
              "type": "string",
            },
          },
          {
            "in": "query",
            "name": "page[before]",
            "required": false,
            "schema": {
              "type": "string",
            },
          },
          {
            "in": "query",
            "name": "include",
            "required": false,
            "schema": {
              "enum": [
                "other",
                "more",
              ],
              "type": "string",
            },
          },
          {
            "in": "query",
            "name": "filter[name]",
            "required": false,
            "schema": {
              "type": "string",
            },
          },
          {
            "in": "query",
            "name": "filter[name][eq]",
            "required": false,
            "schema": {
              "type": "string",
            },
          },
          {
            "in": "query",
            "name": "filter[name][gte]",
            "required": false,
            "schema": {
              "type": "string",
            },
          },
          {
            "in": "query",
            "name": "filter[name][gt]",
            "required": false,
            "schema": {
              "type": "string",
            },
          },
          {
            "in": "query",
            "name": "filter[name][lt]",
            "required": false,
            "schema": {
              "type": "string",
            },
          },
          {
            "in": "query",
            "name": "filter[name][lte]",
            "required": false,
            "schema": {
              "type": "string",
            },
          },
          {
            "in": "query",
            "name": "filter[name][ne]",
            "required": false,
            "schema": {
              "type": "string",
            },
          },
        ],
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "properties": {
                    "data": {
                      "items": {
                        "properties": {
                          "attributes": {
                            "properties": {
                              "createdAt": {
                                "type": "string",
                              },
                              "name": {
                                "type": "string",
                              },
                              "otherId": {
                                "type": "string",
                              },
                            },
                            "required": [
                              "name",
                              "createdAt",
                              "otherId",
                            ],
                            "type": "object",
                          },
                          "id": {
                            "format": "uuid",
                            "type": "string",
                          },
                          "meta": {
                            "properties": {},
                            "type": "object",
                          },
                          "relationships": {
                            "properties": {},
                            "type": "object",
                          },
                          "type": {
                            "enum": [
                              "item",
                            ],
                            "type": "string",
                          },
                        },
                        "required": [
                          "id",
                          "type",
                          "attributes",
                        ],
                        "type": "object",
                      },
                      "type": "array",
                    },
                    "included": {
                      "not": {},
                    },
                    "jsonapi": {
                      "properties": {
                        "profile": {
                          "items": {
                            "type": "string",
                          },
                          "type": "array",
                        },
                        "version": {
                          "enum": [
                            "1.1",
                          ],
                          "type": "string",
                        },
                      },
                      "required": [
                        "profile",
                        "version",
                      ],
                      "type": "object",
                    },
                    "links": {
                      "properties": {
                        "next": {
                          "anyOf": [
                            {
                              "format": "uri",
                              "type": "string",
                            },
                            {
                              "type": "null",
                            },
                          ],
                        },
                        "prev": {
                          "anyOf": [
                            {
                              "format": "uri",
                              "type": "string",
                            },
                            {
                              "type": "null",
                            },
                          ],
                        },
                        "self": {
                          "format": "uri",
                          "type": "string",
                        },
                      },
                      "required": [
                        "self",
                        "next",
                        "prev",
                      ],
                      "type": "object",
                    },
                    "meta": {
                      "count": {
                        "type": "integer",
                      },
                    },
                    "relationships": {
                      "properties": {},
                      "type": "object",
                    },
                  },
                  "required": [
                    "jsonapi",
                    "links",
                    "meta",
                    "data",
                  ],
                  "type": "object",
                  Symbol(json-schema-resolver.ignore): true,
                },
              },
            },
            "description": "Default Response",
          },
        },
      },
    },
    "/items/{id}": {
      "get": {
        "parameters": [
          {
            "in": "path",
            "name": "id",
            "required": true,
            "schema": {
              "type": "string",
            },
          },
        ],
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "properties": {
                    "data": {
                      "properties": {
                        "attributes": {
                          "properties": {
                            "name": {
                              "type": "string",
                            },
                          },
                          "required": [
                            "name",
                          ],
                          "type": "object",
                        },
                        "id": {
                          "format": "uuid",
                          "type": "string",
                        },
                        "relationships": {
                          "properties": {
                            "other": {
                              "properties": {
                                "data": {
                                  "properties": {
                                    "id": {
                                      "format": "uuid",
                                      "type": "string",
                                    },
                                    "type": {
                                      "enum": [
                                        "other",
                                      ],
                                      "type": "string",
                                    },
                                  },
                                  "required": [
                                    "id",
                                    "type",
                                  ],
                                  "type": "object",
                                },
                                "links": {
                                  "properties": {
                                    "self": {
                                      "format": "uri",
                                      "type": "string",
                                    },
                                  },
                                  "required": [
                                    "self",
                                  ],
                                  "type": "object",
                                },
                              },
                              "required": [
                                "data",
                                "links",
                              ],
                              "type": "object",
                            },
                          },
                          "required": [
                            "other",
                          ],
                          "type": "object",
                        },
                        "type": {
                          "enum": [
                            "item",
                          ],
                          "type": "string",
                        },
                      },
                      "required": [
                        "id",
                        "type",
                        "attributes",
                        "relationships",
                      ],
                      "type": "object",
                    },
                    "included": {
                      "not": {},
                    },
                    "jsonapi": {
                      "properties": {
                        "profile": {
                          "items": {
                            "type": "string",
                          },
                          "type": "array",
                        },
                        "version": {
                          "enum": [
                            "1.1",
                          ],
                          "type": "string",
                        },
                      },
                      "required": [
                        "profile",
                        "version",
                      ],
                      "type": "object",
                    },
                    "links": {
                      "properties": {
                        "related": {
                          "format": "uri",
                          "type": "string",
                        },
                        "self": {
                          "format": "uri",
                          "type": "string",
                        },
                      },
                      "required": [
                        "self",
                      ],
                      "type": "object",
                    },
                    "meta": {
                      "properties": {},
                      "type": "object",
                    },
                  },
                  "required": [
                    "jsonapi",
                    "data",
                    "links",
                  ],
                  "type": "object",
                  Symbol(json-schema-resolver.ignore): true,
                },
              },
            },
            "description": "Default Response",
          },
        },
      },
    },
    "/throws-error": {
      "get": {
        "responses": {
          "200": {
            "description": "Default Response",
          },
        },
      },
    },
  },
}
`;
